analysis/deathknightblood/src/modules/features/BoneShieldTimesByStacks.js

```diff
@@ -58,6 +104,28 @@ class BoneShieldStacksBySeconds extends Analyzer {
     });
     return avgStacks;
   }
+
+  reduceDRWCooldown(stackDiff) {
+    if (!this.selectedCombatant.hasLegendaryByBonusID(SPELLS.CRIMSON_RUNE_WEAPON.bonusID)) {
```

Comment:
rather than repeatedly checking this, check it in the constructor and then disable the whole module if the legendary is missing (if the module is only intended to be active with CRW equipped). for example:

```js
if(!this.selectedCombatant.hasLegendary(SPELLS.CRIMSON_RUNE_WEAPON)) {
   this.active = false;
   return;
}
```
before registering your event listeners.

Reply:
Bone shield buff management is relevant regardless of talent/legendary, dancing rune weapon cooldown is added when you have the CRW legendary on. I was copying the pattern from https://github.com/WoWAnalyzer/WoWAnalyzer/blob/shadowlands/analysis/deathknightblood/src/modules/runicpower/RunicPowerTracker.js where cooldown reduction is tracked for vamp blood if red thirst talent is selected. If you'd rather see the CDR tracking from CRW in a module of its own I could do that instead.

Reply:
:+1: that is fine, it just looked like it *might* be turning off (effectively) all the logic of the module here---if it isn't then how you have it is good